cmd/generic-handlers.go

```diff
@@ -401,9 +401,18 @@ func (h httpStatsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// Wraps w to record http response information
 	ww := &httpResponseRecorder{ResponseWriter: w}
 
+	// Time start before the call is about to start.
+	tBefore := UTCNow()
+
 	// Execute the request
 	h.handler.ServeHTTP(ww, r)
 
+	// Time after call has completed.
+	tAfter := UTCNow()
+
+	// Time duration since the call started.
+	duration := tAfter.Sub(tBefore)
+
 	// Update http statistics
-	globalHTTPStats.updateStats(r, ww)
+	globalHTTPStats.updateStats(r, ww, duration.Seconds())
```

Comment:
Can you add a comment saying that `duration.Seconds()` returns a floating and non rounded seconds ?